### DISCUSSION TITLE
Cannot specify id value in select_static()

### DISCUSSION THREAD

**Discussion by s-ohnishi** [phalcon/cphalcon#16093] [Q&A] [answered]
I'm trying Phalcon 5.0.0RC4.
Since Phalcon\Tag will be migrated to Phalcon\Html\TagFactory, it may not be a topic that needs to be dug up now, but I would like to ask.
It seems that select is not yet implemented in TagFactory.

`{{ select_static(['idBoo', 'name':'boo', 'value':'foo', 'class':'someClass'], optionsArray) }}`
is no problem.
`{{ select_static(['dummy', 'id':'idBoo', 'name':'boo', 'value':'foo', 'class':'someClass'], optionsArray) }}`
This is fine too.
`{{ select_static(['id':'idBoo', 'name':'boo', 'value':'foo', 'class':'someClass'], optionsArray) }}`
will result in an error.
(I don't know why, but after waiting for a long time, "Fatal error: Allowed memory size of ~ " is displayed. )

I think Tag::textField() does the same process internally, but there is no problem if you add the 'id' key at the beginning.
`{{ text_field('id':'aa', 'size':30, 'value':'bb') }}`

select_static() requires that the first argument (which will be the id value) be a string without a key, and if you want to clearly indicate the id with a key, you need to put a dummy string at the beginning. Was it designed as a specification?

By the way, "Fatal error: Allowed memory size of ~" every time an error occurs, is xDebug setting wrong?

**Comment by niden** [comment, marked as answer]
The third line results in an error because the method cannot find the first element in the array you are passing. The syntax is:

```php
$id = $parameters[0];
```

You are correct, the first element has to be key-less.

Now you know why we want to move to the TagFactory/helper. ;) We are not there yet, still work to be done especially for select/checkbox/radio but we are getting there

As for the Fatal error etc. I am really not sure. It could very well be something with xDebug internally. I have seen many cases in the past where xDebug will crash Phalcon...

**Reply by s-ohnishi** [reply]
I understand.
That's one of the reasons why you "must" migrate to TagFactory.
But my project won't wait, so I have to use the old Phalcon, even if it's awkward.

Only for select_static(), the first parameter must not be keyed, so maybe this post will be helpful for those who use select_static().
I don't think there is any such explanation.

By the way, it seems that some TagFactory parts can control indentation and line breaks.
I think it would be nice if "all" parts did.

Even if the appearance of the source is irrelevant to the display on the browser, I think it is very good that the HTML document generated by the view is neatly aligned.
I feel that it is effective for development to be able to know immediately that the HTML structure is disturbed.

It's a pity that xDebug and Phalcon may not get along very well.
